webapp/tests/Controller/API/BaseTest.php

```diff
@@ -61,4 +83,165 @@ protected function verifyApiJsonResponse(
+
+        foreach ($this->expectedAbsent as $expectedAbsentId) {
+            $object = null;
+            foreach ($objects as $potentialObject) {
+                if ($potentialObject['id'] == $expectedAbsentId) {
+                    $object = $potentialObject;
+                    break;
+                }
+            }
+
+            static::assertNull($object);
+        }
+    }
+
+    /**
+     * Test that the list action returns the expected objects if ID's are passed
+     */
+    public function testListWithIds()
+    {
+        if ($this->apiEndpoint === null) {
+            static::markTestSkipped('No endpoint defined');
+        }
+        $objects = $this->verifyApiJsonResponse('GET', "/contests/2/$this->apiEndpoint?" . http_build_query(['ids' => array_keys($this->expectedObjects)]), 200, $this->apiUser);
+
+        // Verify we got exactly enough objects
+        static::assertIsArray($objects);
+        static::assertCount(count($this->expectedObjects), $objects);
+
+        // Verify all objects are present
+        foreach (array_keys($this->expectedObjects) as $expectedObjectId) {
+            $object = null;
+            foreach ($objects as $potentialObject) {
+                if ($potentialObject['id'] == $expectedObjectId) {
+                    $object = $potentialObject;
+                    break;
+                }
+            }
+
+            static::assertNotNull($object);
+        }
+    }
+
+    /**
+     * Test that the list action returns the correct error when the contest is not found
+     */
+    public function testListContestNotFound()
+    {
+        if ($this->apiEndpoint === null) {
+            static::markTestSkipped('No endpoint defined');
+        }
+        $respone = $this->verifyApiJsonResponse('GET', "/contests/3/$this->apiEndpoint", 404, $this->apiUser);
+        static::assertEquals('Contest with ID \'3\' not found', $respone['message']);
+    }
+
+    /**
+     * Test that the list method returns the correct error when the ids parameter is not an array
+     */
+    public function testListWithIdsNotArray()
+    {
+        if ($this->apiEndpoint === null) {
+            static::markTestSkipped('No endpoint defined');
+        }
+        $response = $this->verifyApiJsonResponse('GET', "/contests/2/$this->apiEndpoint?" . http_build_query(['ids' => 2]), 400, $this->apiUser);
+        static::assertEquals("'ids' should be an array of ID's to fetch", $response['message']);
+    }
+
+    /**
+     * Test that the list method returns the correct error when passing ID's that don't exist
+     */
+    public function testListWithAbsentIds()
+    {
+        if ($this->apiEndpoint === null) {
+            static::markTestSkipped('No endpoint defined');
+        }
+        $ids      = array_merge(array_keys($this->expectedObjects), $this->expectedAbsent);
+        $response = $this->verifyApiJsonResponse('GET', "/contests/2/$this->apiEndpoint?" . http_build_query(['ids' => $ids]), 404, $this->apiUser);
```

Comment:
I had an sprintf before but I had comments on older changes that people were not a fan of it, but that was maybe for performance reasons. I will re-add it here since for tests that doesn't matter and I like it more myself as well.
Or do you prefer concatenation over sprintf?